go/tasks/plugins/k8s/spark/spark.go

```diff
@@ -112,6 +113,11 @@ func (sparkResourceHandler) BuildResource(ctx context.Context, taskCtx pluginsCo
 		sparkEnvVars[envVar.Name] = envVar.Value
 	}
 
+	if taskCtx.TaskExecutionMetadata().GetOverrides().GetRetryStrategy() != nil {
+		sparkEnvVars["FLYTE_MAX_ATTEMPTS"] = strconv.Itoa(int(taskCtx.TaskExecutionMetadata().GetOverrides().GetRetryStrategy().Retries))
```

Comment:
why do you need these env variables? Initially when we discussed we talked that you want metrics on retries. Adding env variables may add things to the contract. What is the goal of these variables?

Reply:
if we want to add env vars -- we should add it here, as this is the contract that will be maintained.
https://github.com/lyft/flyteplugins/blob/master/go/tasks/pluginmachinery/flytek8s/k8s_resource_adds.go#L36,

Reply:
@kumare3 by exposing the retry number we can do canary based on retryNumber. I have updated the attemptNumber in k8s_resource_adds. The only left problem is related to maxattempt (which I do not think is yet implemented on the propeller/admin side).

Reply:
you are right Max attempt number is not available in the ID at the moment. @akhurana001 and you can work on that together?

Reply:
Will work the max attempt as part of a separate PR under the same task